[Jordan Chua] ip

JordanChua opened the pull request:
## Duke
> “Your mind is for having ideas, not holding them.” – David Allen

Duke frees your mind of  of having to remember things u need to, its 
1. *text-based*
2. ***VERY*** user friendly
3. ~~*USEFUL*~~ ***SUPER USEFUL*** at organising your tasks

All you need to do is simply follow these 4 steps:
1. download it from [here](https://github.com/JordanChua/ip/releases/tag/A-Jar)
2. double click the application
3. start adding your tasks!
4. let it organise and update your tasks for you! 😉

And the best thing is that its **FREE!**
So what are you waiting for? Download it **NOW** and you won't have to worry about managing your tasks!

Our current features are:
- [x] Managing Tasks/Deadlines/Events
- [x] Letting the user know upcoming tasks due within the week
- [ ] Broadening the input syntax for users so it is more accessible
- [ ] Add delight to the experience when all tasks for the week are complete :tada:

Commit:
Add Gradle support

JordanChua committed:
Level-1

JordanChua committed:
Level-1

JordanChua committed:
Merge branch 'master' of https://github.com/JordanChua/ip

JordanChua committed:
Revert "Level-1"

This reverts commit 8d450f6d4572968916ab991b94c4369b345e0b31.

JordanChua committed:
Level-1

JordanChua committed:
Level-2

JordanChua committed:
Level-3

JordanChua committed:
Level-4

JordanChua committed:
A-TextUiTesting

JordanChua committed:
Level-5

JordanChua committed:
Level-6

JordanChua committed:
Level-7

JordanChua committed:
Create duke.txt

JordanChua committed:
Revert "Create duke.txt"

This reverts commit 502fb83f76f748786576aafbed7453a44a9accc1.

JordanChua committed:
Revert "Level-7"

This reverts commit 678717b3dfaed910b9533604372de7f746aa4b7b.

JordanChua committed:
Level-7

JordanChua committed:
Level-8

JordanChua committed:
revert merge

JordanChua committed:
Merge branch 'branch-Level-7'

JordanChua committed:
Merge branch 'branch-Level-8'

JordanChua committed:
Update Duke.java

JordanChua committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

JordanChua committed:
A-MoreOOP

JordanChua committed:
A-Packages

JordanChua committed:
A-JUnit

JordanChua committed:
A-JavaDoc

JordanChua committed:
A-CodingStandard

JordanChua committed:
Level-9

JordanChua committed:
Merge branch 'branch-A-JavaDoc'

JordanChua committed:
Merge branch 'branch-A-CodingStandard'

JordanChua committed:
Merge branch 'branch-Level-9'

JordanChua committed:
Update Parser.java

JordanChua committed:
Update duke.txt

ou4jairmeiogusrjejiepr commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,112 @@
+                        //again assign the date and time separately to local variables
+                        String[] splitTime = dateTime.split("\\s+");
+                        LocalDate date = LocalDate.parse(splitTime[1]
+                                , DateTimeFormatter.ofPattern("dd/MM/yyyy"));
```
Consider breaking the line after the comma instead

ou4jairmeiogusrjejiepr commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,112 @@
+                        LocalDate date = LocalDate.parse(splitTime[1]
+                                , DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+                        LocalTime time = LocalTime.parse(splitTime[2]
+                                , DateTimeFormatter.ofPattern("HHmm"));
```
Consider breaking the line after the comma instead

ou4jairmeiogusrjejiepr commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+public class TaskList {
+    private int numOfTasks;
+    private static List<Task> tasks;
+    private String indent = "    ";
```
Perhaps you can set indent to be private static final

ou4jairmeiogusrjejiepr commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,44 @@
+public class ParserTest {
+
+    @Test
+    void test1() throws DukeException {
```
Perhaps you can consider naming the test cases according to what they are testing

ou4jairmeiogusrjejiepr reviewed (commented):
Code looks good, just a few minor naming details!

JordanChua committed:
Level-10

JordanChua committed:
Level-10

JordanChua committed:
Changed image format to png

JordanChua committed:
Update MainWindow.java

asaierika commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+
+    void add(Task task) {
+        tasks.add(task);
+        System.out.println(indent + "Got it. I've added this task:");
```
Maybe you could use Ui class to print all the messages.

asaierika commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,127 @@
+     * @return command type
+     * @throws DukeException if todo command has no description or the command is not meaningful
+     */
+    static String getCommandType(String input) throws DukeException {
```
I like how you use switch that makes the code look neat.

asaierika commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,96 @@
+                //we get the commandType to know how to process the command
+                String commandType = Parser.getCommandType(fullCommand);
+
+                if (commandType.equals("EXIT")) {
```
Maybe you could use switch here as well.

asaierika commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+    static void printUpcomingTasks(TaskList tasks) {
```
Maybe it would be more suitable to put this method in the TaskList class instead?

asaierika reviewed (commented):
Looks good!
Nice implementation of OOP.

JordanChua committed:
Update Duke.java

JordanChua committed:
Merge branch 'branch-Level-10'

JordanChua committed:
A-Assertion

Add assertions at appropriate places and Update Parser methods

Assertions at appropriate places allow users to identify potential errors and where to find them.

A more meaningful error message when users input the date and time is more useful since users cannot be expected to remember the exact format

Let's,
- add assertions at specific places where certain values are needed
- update ParseDateTime() method to include an error message telling users the correct format

JordanChua committed:
A-CodeQuality

getResponse : replace if else statements with a switch statement

getResponse method has many nested if else statements.

The nested if else statements cause a drop in code readability

Using a switch statement clearly separates the different cases and increases code readability since it is looks neater overall.

JordanChua committed:
Update Ui.java

JordanChua committed:
Merge pull request #1 from JordanChua/branch-A-Assertions

A-Assertion

JordanChua committed:
Merge branch 'master' of https://github.com/JordanChua/ip

* 'master' of https://github.com/JordanChua/ip:
  A-Assertion

JordanChua committed:
Merge branch 'master' into branch-A-CodeQuality

JordanChua committed:
Merge pull request #2 from JordanChua/branch-A-CodeQuality

Branch-A-CodeQuality

JordanChua committed:
Merge branch 'master' of https://github.com/JordanChua/ip

* 'master' of https://github.com/JordanChua/ip:
  Update Ui.java
  A-CodeQuality

JordanChua committed:
Updated Code Quality

JordanChua committed:
C-Sort

Added the extension that allows users to sort deadlines/events in chronological order

JordanChua committed:
Update TaskList

JordanChua committed:
Updated GUI to show message on initialization

JordanChua committed:
Update Readme.md

JordanChua committed:
Set title for chat bot

JordanChua committed:
Merge branch 'master' of https://github.com/JordanChua/ip

JordanChua committed:
Update README.md

JordanChua committed:
Updated README and added UI of chat bot

JordanChua committed:
Merge branch 'master' of https://github.com/JordanChua/ip

JordanChua committed:
Update README.md

JordanChua committed:
Update README.md

JordanChua committed:
Update README.md

JordanChua committed:
Update README.md

JordanChua committed:
Update README.md

JordanChua committed:
Updated Javadocs

JordanChua committed:
Update build.gradle

JordanChua committed:
Merge branch 'master' of https://github.com/JordanChua/ip

JordanChua committed:
Update Duke.java

JordanChua committed:
Fixed bug where tasks were not saved properly

JordanChua committed:
Packaged classes into separate packages

JordanChua committed:
Fixed packaging issues and Updated store method of Storage

JordanChua committed:
Fixed UI packaging

JordanChua committed:
Fixed bug where output is cut off if too long

JordanChua committed:
Updated GUI

JordanChua committed:
Update README.md

JordanChua committed:
Updated UI png

JordanChua committed:
Merge branch 'master' of https://github.com/JordanChua/ip
